src/data/sum/interval.lean

```diff
@@ -0,0 +1,160 @@
+/-
+Copyright (c) 2022 Yaël Dillies. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies
+-/
+import data.sum.order
+import order.locally_finite
+
+/-!
+# Finite intervals in a disjoint union
+
+This file provides the `locally_finite_order` instance for the disjoint sum of two orders.
+
+## TODO
+
+Do the same for the lexicographic sum of orders.
+-/
+
+open function sum
+
+namespace finset
+variables {α₁ α₂ β₁ β₂ γ₁ γ₂ : Type*}
+
+section sum_lift₂
+variables (f f₁ g₁ : α₁ → β₁ → finset γ₁) (g f₂ g₂ : α₂ → β₂ → finset γ₂)
+
+/-- Lifts maps `α₁ → β₁ → finset γ₁` and `α₂ → β₂ → finset γ₂` to a map
+`α₁ ⊕ α₂ → β₁ ⊕ β₂ → finset (γ₁ ⊕ γ₂)`. Could be generalized to alternative monads if we can make
```

Comment:
I'm not sure if "alternative" is a good term to use here, since it could be referring to monads which are alternative functors, or just meaning "other". Could you rephrase this, to clarify what is intended?

Reply:
Rereading this, I don't think I even need `monad`. An alternative functor should be enough.

Reply:
I don't think this resolves the issue, the ambiguity is still present. How about writing "`alternative` functors" to make it clear that you mean `alternative` as a technical term?